### ISSUE TITLE
Data quality: Cases with extremely long case names need remedy

### ISSUE THREAD

**Issue by freelawbot** [freelawproject/courtlistener#247]
The queries below indicate that we currently have some cases in the system with extremely long case names. These won't be easy to fix in an automated way, but would be a great task for a volunteer to work on (or to hand off to something like mechanical turk or MobileWorks). Note that we have more than 150 cases with names longer than 5,000 characters, and more than 3,000 with names longer than 1,000!

Filing in case somebody wants to take on this Herculean task.

 mysql> select count(citationUUID) from Citation where length(case_name) > 15000;
 +---------------------+
 | count(citationUUID) |
 +---------------------+
 |                  11 |
 +---------------------+
 1 row in set (0.53 sec)

 mysql> select count(citationUUID) from Citation where length(case_name) > 10000;
 +---------------------+
 | count(citationUUID) |
 +---------------------+
 |                  24 |
 +---------------------+
 1 row in set (5.07 sec)

 mysql> select count(citationUUID) from Citation where length(case_name) > 5000;
 +---------------------+
 | count(citationUUID) |
 +---------------------+
 |                 156 |
 +---------------------+
 1 row in set (0.72 sec)

 mysql> select count(citationUUID) from Citation where length(case_name) > 1000;
 +---------------------+
 | count(citationUUID) |
 +---------------------+
 |                3062 |
 +---------------------+
 1 row in set (0.49 sec)

 mysql> select count(citationUUID) from Citation where length(case_name) > 500;
 +---------------------+
 | count(citationUUID) |
 +---------------------+
 |               10973 |
 +---------------------+
 1 row in set (0.51 sec)

tag: cleanup, data quality

---
- Bitbucket: https://bitbucket.org/mlissner/search-and-awareness-platform-courtlistener/issue/217
- Originally Reported By: [Mike Lissner](http://bitbucket.org/mlissner)
- Originally Created At: 2012-04-20T19:07:38.609

**Comment by mlissner** [comment]
We'll be fixing a bunch of these as part of #350. Worth re-running after that's done to see how much progress we've made.

**Comment by mlissner** [comment]
Worked on this a bit today to get it where we could have a data entry specialist work on fixing it. The query I used to get a list of long cases is:

    long_names = OpinionCluster.objects.annotate(
        cn_len=Length('case_name_full')
    ).filter(
        cn_len__gte=500, 
        case_name='', 
        source__icontains='r'
    ).order_by(
        Length('case_name_full').desc()
    ).distinct()

This brings back about 10k items with really long case names that we got from resource.org with longest items coming first. There were two from SCOTUS which I fixed manually, the rest need manual love, and I've attached a spreadsheet of them here.
[long-case-names.xlsx](https://github.com/freelawproject/courtlistener/files/1045752/long-case-names.xlsx)
